group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/consumer/ConsumerGroup.java

```diff
@@ -342,6 +348,46 @@ private void removeStaticMember(ConsumerGroupMember oldMember) {
         }
     }
 
+    /**
+     * Update the resolved regular expression.
+     *
+     * @param regex                         The regular expression.
+     * @param newResolvedRegularExpression  The regular expression's metadata.
+     */
+    public void updateResolvedRegularExpression(
+        String regex,
+        ResolvedRegularExpression newResolvedRegularExpression
+    ) {
+        removeResolvedRegularExpression(regex);
+        if (newResolvedRegularExpression != null) {
+            resolvedRegularExpressions.put(regex, newResolvedRegularExpression);
+            newResolvedRegularExpression.topics.forEach(topicName -> subscribedTopicNames.compute(topicName, Utils::incValue));
```

Comment:
doing this here means that a regex received in a HB will be consider as having no subscribers until we resolve it (and replay), vs. inc the counter when we receive the HB. Is this intentional? I see the point on both approaches.. just thinking if this "gap" in time for accounting the subscriber could have some side-effects...

Reply:
We have two data structures:
1) `subscribedRegularExpressions` which is updated when the member subscribed regex is updated. Hence here we account for the change immediately.
2) `resolvedRegularExpressions` which is updated when the regular is resolved. At the same time, we update `subscribedTopicNames` to take into account the subscribed topics via the regex. We can only do this when the regex is resolved though.